src/Mutation/FilteredCollection.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Softonic\GraphQL\Mutation;
+
+use Softonic\GraphQL\Traits\JsonSerializer;
+
+class FilteredCollection implements MutationObject, \JsonSerializable
+{
+    use JsonSerializer;
+
+    protected $arguments = [];
+
+    protected $config;
+
+    private $hasChanged = false;
+
+    public function __construct(array $arguments = [], array $config = [], bool $hasChanged = false)
+    {
+        $this->arguments  = $arguments;
+        $this->config     = $config;
+        $this->hasChanged = $hasChanged;
+    }
+
+    public function __get(string $key)
+    {
+        $items = [];
+
+        foreach ($this->arguments as $argument) {
+
+            $items[] = $argument->{$key};
+        }
+
+        return new Collection($items, $this->config[$key]->children);
+    }
+
+    public function set(array $data)
+    {
+        foreach ($this->arguments as $argument) {
+            if ($argument instanceof Collection) {
+                $argument->set($data);
+            } else {
+                $this->setItemData($argument, $data);
+            }
+        }
+    }
+
+    public function filter(array $filters)
+    {
+        $filteredData = [];
+        if (!empty($this->arguments[0]) && $this->arguments[0] instanceof Collection) {
```

Comment:
Is the first item only checked because all items must be of the same type? Can we extract this condition to make it clearer?